runtime/lua/vim/shared.lua

```diff
@@ -457,6 +457,32 @@ function vim.tbl_flatten(t)
   return result
 end
 
+--- Enumerate a table sorted by its keys.
+---
+---@see Based on https://github.com/premake/premake-core/blob/master/src/base/table.lua
+---
+---@param t table List-like table
+---@return table key-sorted copy of the given table
+function vim.spairs(t)
+  assert(type(t) == 'table', string.format('Expected table, got %s', type(t)))
+
+  -- collect the keys
+  local keys = {}
+  for k in pairs(t) do
+    table.insert(keys, k)
+  end
+  table.sort(keys)
+
+  -- return the iterator function
+  local i = 0
+  return function()
+    i = i + 1
+    if keys[i] then
+      return keys[i], t[keys[i]]
```

Comment:
should we try to be more like ipairs/pairs here? From `:help luaref-pairs()` : 

```
pairs({t})                                                      *luaref-pairs()*
        Returns three values: the `next` |luaref-next()| function, the table
        {t}, and `nil`, so that the construction

               for k,v in pairs(t) do  body  end

        will iterate over all key-value pairs of table {t}.
```

i.e. should we return "iterator function, table {t}, and `nil`"? Then need some
test coverage for `for k,v in pairs(t) do  body  end`.

Reply:
I guess so - for now I just kept it consistent with the implementation from premake. Maybe something to improve later?

Reply:
i suppose the door is open for changing this in the future, let's at least leave a TODO comment in the source